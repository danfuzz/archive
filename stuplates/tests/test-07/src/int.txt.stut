== true
${ (:int: 0 == 0) }
${ (:int: 10 == 10) }

== false
${ (:int: 0 == 1) }
${ (:int: 100 == -100) }

!= true
${ (:int: 0 != 1) }
${ (:int: 100 != -100) }

!= false
${ (:int: 0 != 0) }
${ (:int: 10 != 10) }

< true
${ (:int: 5 < 6) }
${ (:int: 18 < 3600) }

< false
${ (:int: 5 < -5) }
${ (:int: -10 < -20) }

> true
${ (:int: 2 > -2) }
${ (:int: 100 > 2) }

> false
${ (:int: 4 > 5) }
${ (:int: -1 > 10) }

<= true
${ (:int: 0 <= 0) }
${ (:int: 0 <= 10) }

<= false
${ (:int: 0 <= -1) }
${ (:int: 5 <= 2) }

>= true
${ (:int: 20 >= 20) }
${ (:int: -20 >= -22) }

>= false
${ (:int: 1 >= 5) }
${ (:int: 0 >= 50) }

+
${ (:int: 0 + 0) } 0
${ (:int: 1 + 3) } 4
${ (:int: +"05") } 5
${ (:int: +"-20") } -20
${ (:int: +5.3) } 5
${ (:int: +5.8) } 5
${ def x := -5.3; (:int: +x) } -5
${ x := -5.8; (:int: +x) } -5
${ (:int: +true) } 1
${ (:int: +false) } 0

-
${ (:int: 0 - 0) } 0
${ (:int: 2 - 5) } -3
${ (:int: -3) } -3
${ (:int: -0) } 0

*
${ (:int: 33 * 2) } 66
${ (:int: 666 * 0) } 0
${ (:int: 32 * -2) } -64

/
${ (:int: 15 / 2) } 7
${ (:int: 16 / 2) } 8
${ (:int: -8 / 8) } -1

%
${ (:int: 15 % 2) } 1
${ (:int: 16 % 2) } 0
${ (:int: -8 % 8) } 0
${ (:int: -15 % 2) } -1
${ (:int: -15 % -2) } -1

%%
${ (:int: 15 %% 2) } 1
${ (:int: 16 %% 2) } 0
${ (:int: -8 %% 8) } 0
${ (:int: -15 %% 2) } 1

**
${ (:int: 1 ** 10) } 1
${ (:int: 2 ** 8) } 256

&
${ (:int: 1 & 1) } 1
${ (:int: 16 & 8) } 0
${ (:int: 123 & 456) } 72
${ (:int: 5400 & -1) } 5400
${ (:int: 5400 & 0) } 0

|
${ (:int: 1 | 1) } 1
${ (:int: 16 | 8) } 24
${ (:int: 123 | 456) } 507
${ (:int: 5400 | -1) } -1
${ (:int: 5400 | 0) } 5400

^
${ (:int: 1 ^ 1) } 0
${ (:int: 16 ^ 8) } 24
${ (:int: 123 ^ 456) } 435
${ (:int: 5400 ^ -1) } -5401
${ (:int: 5400 ^ 0) } 5400

<<
${ (:int: 1 << 0) } 1
${ (:int: 23 << 1) } 46
${ (:int: 64 << -1) } 32
${ (:int: 7 << -2) } 1

>>
${ (:int: 1 >> 0) } 1
${ (:int: 23 >> 1) } 11
${ (:int: 64 >> 2) } 16
${ (:int: 7 >> -1) } 14

~
${ (:int: ~5) } -6
${ (:int: ~ -1) } 0
