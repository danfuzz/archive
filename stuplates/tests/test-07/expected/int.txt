== true
true
true

== false
false
false

!= true
true
true

!= false
false
false

< true
true
true

< false
false
false

> true
true
true

> false
false
false

<= true
true
true

<= false
false
false

>= true
true
true

>= false
false
false

+
0 0
4 4
5 5
-20 -20
5 5
5 5
-5 -5
-5 -5
1 1
0 0

-
0 0
-3 -3
-3 -3
0 0

*
66 66
0 0
-64 -64

/
7 7
8 8
-1 -1

%
1 1
0 0
0 0
-1 -1
-1 -1

%%
1 1
0 0
0 0
1 1

**
1 1
256 256

&
1 1
0 0
72 72
5400 5400
0 0

|
1 1
24 24
507 507
-1 -1
5400 5400

^
0 0
24 24
435 435
-5401 -5401
5400 5400

<<
1 1
46 46
32 32
1 1

>>
1 1
11 11
16 16
14 14

~
-6 -6
0 0
